packages/Utils/test/tstDBC.cpp

```diff
@@ -304,6 +304,31 @@ TEUCHOS_UNIT_TEST( DataTransferKitException, errorcode_test_2 )
     }
 }
 
+//---------------------------------------------------------------------------//
+// Check that we can catch a
+// DataTransferKit::DataTransferKitNotImplementedException and that the
+// appropriate error message is written.
+TEUCHOS_UNIT_TEST( DataTransferKitNotImplementedException, message_error )
+{
+    std::string message;
+    try
+    {
+        throw DataTransferKit::DataTransferKitNotImplementedException();
+    }
+    catch ( const DataTransferKit::DataTransferKitNotImplementedException
+                &assertion )
+    {
+        message = std::string( assertion.what() );
+    }
+    catch ( ... )
+    {
+        TEST_ASSERT( 0 );
+    }
```

Comment:
Use `TEST_THROW()` macro from `Teuchos` unit testing framework.

Reply:
On second thoughts, this is good enough for now.  We can change whenever we revisit exceptions in DTK.